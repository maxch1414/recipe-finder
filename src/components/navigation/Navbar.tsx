import React from "react";
import Link from "next/link";
import { Button } from "../ui/button";
import {
  Sheet,
  SheetTrigger,
  SheetContent,
  SheetHeader,
  SheetTitle,
  SheetDescription,
} from "@/components/ui/sheet";
import { Menu } from "lucide-react";
import Image from "next/image";

export function Navbar() {
  return (
    <header className="bg-primary text-primary-foreground">
      <div className="container mx-auto px-4">
        <nav className="flex items-center justify-between py-4">
          <Link href="/" className="text-2xl font-bold flex items-center">
            <Image
              src="/logo.svg"
              alt="Recipe Finder"
              width={30}
              height={30}
              className="mr-2"
            />
            Recipe Finder
          </Link>
          <ul className="hidden md:flex space-x-6">
            <li>
              <Link href="/about" className="hover:underline">
                About
              </Link>
            </li>
            <li>
              <Link href="/contact" className="hover:underline">
                Contact
              </Link>
            </li>
          </ul>
          <Sheet>
            <SheetTrigger asChild>
              <Button variant="outline" size="icon" className="md:hidden">
                <Menu className="h-6 w-6" />
                <span className="sr-only">Open menu</span>
              </Button>
            </SheetTrigger>
            <SheetContent side="right">
              <SheetHeader>
                <SheetTitle>Menu</SheetTitle>
                <SheetDescription>Navigate through our app</SheetDescription>
              </SheetHeader>
              <nav className="flex flex-col space-y-4 mt-4">
                <Link href="/about" className="hover:underline">
                  About
                </Link>
                <Link href="/contact" className="hover:underline">
                  Contact
                </Link>
              </nav>
            </SheetContent>
          </Sheet>
        </nav>
      </div>
    </header>
  );
}
