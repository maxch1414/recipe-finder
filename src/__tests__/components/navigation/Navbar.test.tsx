import React from "react";
import { expect, test } from "vitest";
import { render } from "@testing-library/react";

import { Navbar } from "@/components/navigation/Navbar";

test("Renders a ingredient form correctly", async () => {
  const { container } = render(<Navbar />);

  expect(container).toMatchSnapshot();
});
