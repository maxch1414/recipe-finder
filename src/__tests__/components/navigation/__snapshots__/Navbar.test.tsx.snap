// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Renders a ingredient form correctly 1`] = `
<div>
  <header
    class="bg-primary text-primary-foreground"
  >
    <div
      class="container mx-auto px-4"
    >
      <nav
        class="flex items-center justify-between py-4"
      >
        <a
          class="text-2xl font-bold flex items-center"
          href="/"
        >
          <img
            alt="Recipe Finder"
            class="mr-2"
            data-nimg="1"
            decoding="async"
            height="30"
            loading="lazy"
            src="/logo.svg"
            style="color: transparent;"
            width="30"
          />
          Recipe Finder
        </a>
        <ul
          class="hidden md:flex space-x-6"
        >
          <li>
            <a
              class="hover:underline"
              href="/about"
            >
              About
            </a>
          </li>
          <li>
            <a
              class="hover:underline"
              href="/contact"
            >
              Contact
            </a>
          </li>
        </ul>
        <button
          aria-controls="radix-:r0:"
          aria-expanded="false"
          aria-haspopup="dialog"
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground h-9 w-9 md:hidden"
          data-state="closed"
          type="button"
        >
          <svg
            class="lucide lucide-menu h-6 w-6"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="4"
              x2="20"
              y1="12"
              y2="12"
            />
            <line
              x1="4"
              x2="20"
              y1="6"
              y2="6"
            />
            <line
              x1="4"
              x2="20"
              y1="18"
              y2="18"
            />
          </svg>
          <span
            class="sr-only"
          >
            Open menu
          </span>
        </button>
      </nav>
    </div>
  </header>
</div>
`;
